zebra-scan/src/bin/scanner/main.rs

```diff
@@ -0,0 +1,104 @@
+//! The zebra-scanner binary.
+//!
+//! The zebra-scanner binary is a standalone binary that scans the Zcash blockchain for transactions using the given sapling keys.
+use lazy_static::lazy_static;
+use structopt::StructOpt;
+use tracing::*;
+
+use zebra_chain::parameters::Network;
+use zebra_state::{ChainTipSender, SaplingScanningKey};
+
+use core::net::SocketAddr;
+use std::path::PathBuf;
+
+#[tokio::main]
+/// Runs the zebra scanner binary with the given arguments.
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Display all logs from the zebra-scan crate.
+    tracing_subscriber::fmt::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .init();
+
+    // Parse command line arguments.
+    let args = Args::from_args();
+    let network = args.network;
+    let sapling_keys_to_scan = args
+        .sapling_keys_to_scan
+        .into_iter()
+        .map(|key| (key, 1))
```

Comment:
structopt is not accepting a hashmap for `sapling_keys_to_scan` so we are losing birthday height.

Reply:
We can forget about this if we decide to do https://github.com/ZcashFoundation/zebra/pull/8608#discussion_r1639872969

Reply:
I think we should just remove the keys provided at startup altogether, but it would be fine to omit birthday heights. We could also try a different cli arg parser (`clap` should work) or implement `FromStr` for a struct with the birthday height, like:

```rust
#[derive(serde::Deserialize)]
struct SaplingKey {
    key: SaplingScanningKey,
    #[serde(default = "min_height")]
    birthday_height: Height,
}

fn min_height() -> Height {
    Height(0)
}

impl std::str::FromStr for SaplingKey {
    type Err = Box<dyn std::error::Error>;
    fn from_str(value: &str) -> Result<Self, Self::Err> {
        Ok(serde_json::from_str(value)?)
    }
}
```

Reply:
Let's keep it for now, we can always remove later. It should be easier to move some tests over if we keep the feature.

I added the suggested code to have support for birthday heights.